### PR TITLE
feat: use libuv for tempfiles

### PR DESCRIPTION
**Pull request by hargoniX**
This is currently broken because of linker issues. CC @TwoFX

### CODE REVIEW

**Comment by TwoFX** [comment]
I would appreciate if @Kha could sign off on these rather invasive changes.

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -553,7 +602,7 @@ if (${CMAKE_SYSTEM_NAME} MATCHES "Emscripten")
   # simple. (And we are not interested in `Lake` anyway.) To use dynamic
   # linking, we would probably have to set MAIN_MODULE=2 on `leanshared`,
   # SIDE_MODULE=2 on `lean`, and set CMAKE_SHARED_LIBRARY_SUFFIX to ".js".
-  string(APPEND LEAN_EXE_LINKER_FLAGS " ${LIB}/temp/libleanshell.a ${TOOLCHAIN_STATIC_LINKER_FLAGS} ${EMSCRIPTEN_SETTINGS} -lnodefs.js -s EXIT_RUNTIME=1 -s MAIN_MODULE=1 -s LINKABLE=1 -s EXPORT_ALL=1")
+  string(APPEND LEAN_EXE_LINKER_FLAGS " ${LIB}/temp/libleanshell.a ${TOOLCHAIN_STATIC_LINKER_FLAGS} ${EMSCRIPTEN_SETTINGS} -lnodefs.js -s EXIT_RUNTIME=1 -s MAIN_MODULE=1 -s LINKABLE=1 -s EXPORT_ALL=1 -s ERROR_ON_UNDEFINED_SYMBOLS=0")
```
Comment (Kha) [inline src/CMakeLists.txt:605]: needs comment

**Review by Kha — CHANGES_REQUESTED** [review]

File: script/prepare-llvm-linux.sh [hunk 2]
```diff
@@ -48,6 +48,7 @@ $CP llvm-host/lib/*/lib{c++,c++abi,unwind}.* llvm-host/lib/
 $CP -r llvm/include/*-*-* llvm-host/include/
 # glibc: use for linking (so Lean programs don't embed newer symbol versions), but not for running (because libc.so, librt.so, and ld.so must be compatible)!
 $CP $GLIBC/lib/libc_nonshared.a stage1/lib/glibc
+$CP $GLIBC/lib/libpthread_nonshared.a stage1/lib/glibc
```
[the remark this answers is not in the document]
Comment (TwoFX) [reply script/prepare-llvm-linux.sh:51]: This is needed if you use the pthreads function `pthread_atfork` (only then - there is literally only one symbol in that library. As far as I can tell, Lean itself doesn't need this function, but LibUV does.

File: src/CMakeLists.txt (hunk 1 above)
Comment (TwoFX) [reply src/CMakeLists.txt:608]: Done

File: script/prepare-llvm-linux.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (Kha) [reply script/prepare-llvm-linux.sh:51]: Interesting. Let's document this as well?

**Review by Kha — APPROVED** [review]

File: script/prepare-llvm-linux.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (TwoFX) [reply script/prepare-llvm-linux.sh:52]: Done.

File: src/runtime/io.cpp [hunk 3]
```diff
@@ -831,50 +940,42 @@ extern "C" LEAN_EXPORT obj_res lean_io_rename(b_obj_arg from, b_obj_arg to, lean
...
+    ret = uv_fs_mkstemp(NULL, &req, path, NULL);
+    if (ret < 0) {
         // If mkstemp throws an error we cannot rely on path to contain a proper file name.
-        return io_result_mk_error(decode_io_error(errno, nullptr));
+        return io_result_mk_error(decode_uv_error(ret, nullptr));
     } else {
-        FILE* handle = fdopen(fd, "r+");
-        object_ref pair = mk_cnstr(0, io_wrap_handle(handle), mk_string(path));
+        FILE* handle = fdopen(req.result, "r+");
+        object_ref pair = mk_cnstr(0, io_wrap_handle(handle), mk_string(req.path));
```
Comment (eric-wieser) [inline src/runtime/io.cpp:978]: This leaks the memory within `req`; see #7151 for a fix

**Pull request merged**